### PR TITLE
feat/922: Update namadillo for wasm checksums

### LINKED ISSUE #922: Extension: Remove wasm checksums from persisted storage, update client accordingly
In https://github.com/anoma/namada-interface/pull/897, we added the ability to deserialize Tx bytes in the extension. However, since the extension is currently responsible for persisting these once they are fetched, it is possible that a site to provide invalid hashes for a particular chain ID, which would prevent Tx details from displaying.

- Look into a better way of providing these to the extension. It could be added to `connect()`, or simply invoke `addTxWasmHashes(...)` _after_ a successful connection by an interface, but we would remove the `getTxWasmHashes()` call from the extension, meaning the interface would persist these instead
- Update namadillo atom service for checksums to remove the check inside `fetchWasmHashes`, and persist the response from the SDK (soon to be the indexer API)

### LINKED ISSUE #923: SDK/Indexer: Move fetch for wasm checksum hashes into the indexer
In https://github.com/anoma/namada-interface/pull/897, we added an SDK rpc query for wasm checksum hashes. This should also be added to the indexer:

- Introduce a query to fetch wasm code paths & the checksums associated with them from the indexer @ https://github.com/anoma/namada-indexer
- Update `namadillo` to use this instead of the SDK RPC query

### PR DESCRIPTION
**Pull request by jurevans**
Resolves #922 
Resolves #923 (Indexer changes are merged!)

This PR updates the wasm checksum hashes for details to allow a client to optionally pass these to the signer in order to display details. Previously, these were added to the extension and persisted, however, this would allow a site to potentially update invalid hashes which would prevent details from displaying. As it's already the responsibility of the interface to query and provide these, I've moved them to the `.sign()` call invoked on the integration as an optional argument. If they are not provided, then the commitments within a batch will not deserialize, but signing will still work!

- [x] Get checksums from indexer API in chain parameters
- [x] Remove `addTxWasmHashes` & `getTxWasmHashes` from integration, along with all handling in extension
- [x] Add `checksums` to `PendingTx` as optional prop, update messages accordingly
- [x] Update namadillo to provide these checksums when signing - this means that the methods which invoke sign will need to pass along the checksums from the React components where the checksums value is accessed

__NOTE__ the `sign()` API is also changed here: https://github.com/anoma/namada-interface/pull/815, this removed `tx_type` & `wrapper_tx_msg`, the the max num of args ultimately will be 3

### CODE REVIEW

File: packages/types/src/namada.ts [hunk 1]
```diff
@@ -8,10 +8,13 @@ export type SignArbitraryProps = {
 };
 
 export type SignProps = {
+  // TODO: Simplify these props!
+  // Remove txType & wrapperTxMsg!
```
Comment (jurevans) [inline packages/types/src/namada.ts:12]: Ignore, this is done in https://github.com/anoma/namada-interface/pull/815

File: packages/integrations/src/Namada.ts [hunk 2]
```diff
@@ -56,17 +55,4 @@ export default class Namada implements Integration<Account, Signer> {
   public signer(): Signer | undefined {
     return this._namada?.getSigner();
   }
-
-  public async addTxWasmHashes(
```
Comment (emccorson) [inline packages/integrations/src/Namada.ts:60]: Very minor thing, but these are still present in the `types` docs.

**Review by emccorson — APPROVED** [review]
Looks good to me and works nicely!

**Review by mateuszjasiuk — APPROVED** [review]
Tested and works nicely!

File: apps/namadillo/src/lib/query.ts [hunk 3]
```diff
@@ -117,6 +119,10 @@ export const signTx = async <T>(
   const integration = getIntegration(chain.id);
   const signingClient = integration.signer() as Signer;
 
+  const store = createStore();
```
[the remark this answers is not in the document]
Comment (jurevans) [reply apps/namadillo/src/lib/query.ts]: Got it, thanks!

File: packages/integrations/src/Namada.ts (hunk 2 above)
Comment (jurevans) [reply packages/integrations/src/Namada.ts:60]: Thanks! Updated

**Pull request merged**